routemaster/models/event.rb

```diff
@@ -23,6 +23,7 @@ def initialize(**options)
         @data      = EventData.build options[:data]
 
         _assert VALID_TYPES.include?(@type), 'bad event type'
+        _assert options[:timestamp].to_i <= Routemaster.now, 'timestamp is from the future'
```

Comment:
I think that your point is fair, but I can also see a counter argument. Clock drift is something we'd in general want to avoid, particularly for the sake of our logging systems. In a way, this can act as an early warning system for that. 

Secondly, I think it's not that much of an issue with our infra. We're expecting our hosts to be Solid hardware > VM > Container. All of which should have access to NTP, and likely the second two steps will only be alive for fairly short periods, where they'll sync time at boot. We're not in the familiar environment of a handful of old machines we own with knackered CMOS batteries that we have to allow for. 

So I kind of think that it's ok to be tough on clients, because otherwise we're essentially making an allowance for mild hardware issues.